core/sqf/src/stfs/stfs_stub.cpp

```diff
@@ -328,63 +332,69 @@ copySTFSLocation(int pv_instnum)
 {
 
   //Directory locations not specified 
-  if ((STFS_stub::stfsDirsHDD_.numLocations <= 0) && (STFS_stub::overflowType_ == STFS_HDD)) {
+  if ((STFS_stub::stfsDirsHDD_.numLocations <= 0) && (STFS_stub::overflowType_ == STFS_HDD)) 
     return;
-  }
-  if ((STFS_stub::stfsDirsSSD_.numLocations <= 0) && (STFS_stub::overflowType_ == STFS_SSD)) {
+  
+  if ((STFS_stub::stfsDirsSSD_.numLocations <= 0) && (STFS_stub::overflowType_ == STFS_SSD)) 
     return;
-  }
+  
 
 
-  if(STFS_stub::overflowType_ == STFS_HDD) {
-    copyLocationHelper(STFS_stub::stfsDirsHDD_, pv_instnum);
-  }
-  else if(STFS_stub::overflowType_ == STFS_SSD) {
+  if(STFS_stub::overflowType_ == STFS_HDD) 
+    copyLocationHelper(STFS_stub::stfsDirsHDD_, pv_instnum);  
+  else if(STFS_stub::overflowType_ == STFS_SSD) 
     copyLocationHelper(STFS_stub::stfsDirsSSD_, pv_instnum);
-  }
+  
 
-  if (STFS_stub::stfsLocation_[strlen(STFS_stub::stfsLocation_) - 1] != '/') {
+  if (STFS_stub::stfsLocation_[strlen(STFS_stub::stfsLocation_) - 1] != '/') 
     strcat(STFS_stub::stfsLocation_, "/");
-  }
-
+  
 }
 
+/// \brief Sets the scratchdisks as specified in the CQD SCRATCH_DISKS 
+///        and avoids looking at envvar
+void STFS_set_scratch_dirs(char *pv_scratchloc) 
+{
+   memset(STFS_stub::scratchCQDLocationString_, 0, sizeof(STFS_stub::scratchCQDLocationString_));
+  strcpy((char *)STFS_stub::scratchCQDLocationString_,pv_scratchloc);
+}
 /// \brief Checks for the Location Env Variable 
 ///        and processes it
 void STFS_checkLocation(int pv_instnum) 
 {
+  char *lp_HDDEnvLocation = getenv("STFS_HDD_LOCATION");
+  char *lp_SSDEnvLocation = getenv("STFS_SSD_LOCATION");
   if (STFS_stub::locationEnvChecked_) {
     copySTFSLocation(pv_instnum);
     return;
   }
-
-  char *lp_HDDEnvLocation = getenv("STFS_HDD_LOCATION");
-  char *lp_SSDEnvLocation = getenv("STFS_SSD_LOCATION");
-  char *lp_EnvLocation    = getenv("STFS_LOCATION");
-
+  if ((strlen(STFS_stub::scratchCQDLocationString_)>0) && (STFS_stub::overflowType_ == STFS_HDD)) 
```

Comment:
OK will consider this for a future improvement.  We need to have a discussion prior to this to  see if there may be a case where we could temporarily use SSD scratch locations and need the ability to have both types.